lib/client.js

```diff
@@ -1587,6 +1587,8 @@ async function writeIterable ({ client, request, socket, contentLength, header,
 
   let bytesWritten = 0
   try {
+    // TODO (fix): What if socket errors while waiting for body?
```

Comment:
Technically, once the next yield happens, the `for...await of` will take care of it (as it throws socket[kError])

Reply:
Yes, but body could be doing a lot of work before the next yield happens.

Reply:
I tried doing something smarter, but calling `return` doesn't affect the iterator at all until it yields anyway, so it somehow has to be in userland.